### PR TITLE
Fix socat invocation in dnsmasq image 

### PR DESCRIPTION
**Pull request by lepiolet**
## Description

https://github.com/appgate/sdp-k8s-injector/commit/57390d6b3e71f9731704d7ab5e2561b5d66fb326#diff-bc9bd89c20c87435874782d8b85ac7b101f17fc7b8814629e9ca2c071625cb7eL4 this PR made the killing mechanism to fail because some changes in how socat exists.

See https://repo.or.cz/socat.git/blob/2da070164d454971d5c970b5278e645051f0d0f7:/CHANGES

```
In previous Socat versions errors EPIPE and ECONNRESET on read() were handled at warning level, thus not automatically leading to termination with exit code 1.
```

## Checklist
- [x] Bump `.version` and `.appVersion` in [k8s/crd/Chart.yaml](../k8s/crd/Chart.yaml)
- [x] Bump `.version` and `.appVersion` in [k8s/chart/Chart.yaml](../k8s/chart/Chart.yaml)

### CODE REVIEW

File: assets/sdp-service-watcher [hunk 1]
```diff
@@ -13,13 +13,9 @@ respond() {
 }
 
 kill_all() {
-    for f in /var/run/sdp-driver /var/run/sdp-dnsmasq /var/run/sdp-service
+    for f in /var/run/sdp-dnsmasq /var/run/sdp-driver /var/run/sdp-service
```
Comment (thomascellerier) [inline assets/sdp-service-watcher:16]: Why was the order important here?

File: assets/wait-for-stop [hunk 2]
```diff
@@ -1,5 +1,10 @@
...
 
 SOCK_PATH=$1
+if [ -z "${2}" ]; then
+    LOG="Exiting container"
+else
+    LOG="Exiting container $2"
+fi
 echo "Listening to stop service on $SOCK_PATH"
-socat UNIX-LISTEN:"$SOCK_PATH" system:exit,fdout=4
+socat UNIX-LISTEN:"$SOCK_PATH" system:"echo $LOG;sleep 1;exit",fdout=4
```
Comment (thomascellerier) [inline assets/wait-for-stop:10]: Why is the sleep needed here?

**Review by thomascellerier — APPROVED** [review]
Does this require a version bump?

File: assets/sdp-service-watcher (hunk 1 above)
Comment (lepiolet) [reply assets/sdp-service-watcher:16]: it's not really important. I can change it if you prefer. I changed it just to test something

File: assets/wait-for-stop (hunk 2 above)
Comment (lepiolet) [reply assets/wait-for-stop:10]: ```In previous Socat versions errors EPIPE and ECONNRESET on read() were handled at warning level, thus not automatically leading to termination with exit code 1.```

See the CHANGELOG in socat. They have fixed the code not to ignore the EPIPE/ECONNRESET. Note that before we upgraded the image we were also getting the read pipe error. See here:

```
  read(5, 0x7f47b8f4b000, 8192)           = -1 ECONNRESET (Connection reset by peer)
  recvfrom(3, 0x7fff70a81830, 519, MSG_DONTWAIT, NULL, NULL) = -1 EAGAIN (Resource temporarily unavailable)
  shutdown(6, SHUT_WR)                    = 0
  shutdown(6, SHUT_RDWR)                  = 0
  unlink("/tmp/test.sock")                = 0
  shutdown(5, SHUT_RDWR)                  = 0
  munmap(0x7f47b8f4a000, 24576)           = 0
  recvfrom(3, 0x7fff70a82060, 519, MSG_DONTWAIT, NULL, NULL) = -1 EAGAIN (Resource temporarily unavailable)
  exit_group(0)                           = ?
  +++ exited with 0 +++
```


While new socat fails on it

```
  read(5, 0x7f8b346a0000, 8192)           = -1 ECONNRESET (Connection reset by peer)
  open("/etc/localtime", O_RDONLY|O_NONBLOCK|O_LARGEFILE|O_CLOEXEC) = -1 ENOENT (No such file or directory)
  getpid()                                = 14
  writev(2, [{iov_base="", iov_len=0}, {iov_base="2024/09/24 05:42:24 socat[14] E "..., iov_len=88}], 2) = 88
  shutdown(6, SHUT_RDWR)                  = 0
  unlink("/tmp/test.sock")                = 0
  shutdown(5, SHUT_RDWR)                  = 0
  exit_group(1)                           = ?
  +++ exited with 1 +++```

**Comment by lepiolet** [comment]
> Does this require a version bump?

yeah

**Review by thomascellerier — APPROVED** [review]

**Pull request merged**